velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -20,12 +20,81 @@
 #include "velox/connectors/hive/HiveConnector.h"
 #include "velox/connectors/hive/HiveWriteProtocol.h"
 #include "velox/dwio/dwrf/writer/Writer.h"
+#include "velox/exec/OperatorUtils.h"
 
 using namespace facebook::velox::dwrf;
 using WriterConfig = facebook::velox::dwrf::Config;
 
 namespace facebook::velox::connector::hive {
 
+namespace {
+std::vector<column_index_t> getPartitionChannels(
+    std::shared_ptr<const HiveInsertTableHandle> insertTableHandle) {
+  std::vector<column_index_t> channels;
+
+  for (column_index_t i = 0; i < insertTableHandle->inputColumns().size();
+       i++) {
+    if (insertTableHandle->inputColumns()[i]->isPartitionKey()) {
+      channels.push_back(i);
+    }
+  }
+
+  return channels;
+}
+
+RowTypePtr getChannelTypes(
```

Comment:
Like I commented on the other PR, this looks should not be needed. We can pass 'rowType' to PartitionidGenerator directly.

Reply:
Deprecated